### ISSUE TITLE
Zero-based index in `list_reduce`

### ISSUE THREAD

**Issue by soerenwolfers** [duckdb/duckdb#15062]
### What happens?

```sql
SELECT list_reduce(['a', 'b'], (x, y, i) -> x || y || i)
```
```
┌──────────────────────────────────────────────────────────────────────────────────┐
│ list_reduce(main.list_value('a', 'b'), (main."row"(x, y, i) -> ((x || y) || i))) │
│                                     varchar                                      │
├──────────────────────────────────────────────────────────────────────────────────┤
│ ab1                                                                              │
└──────────────────────────────────────────────────────────────────────────────────┘
```
but according to https://duckdb.org/docs/sql/functions/lambda.html 

> All lambda functions accept an optional extra parameter that represents the index of the current element. This is always the last parameter of the lambda function, and is 1-based (i.e., the first element has index 1).

which strongly suggests that `y = 'b'` should be accompanied with `i = 2`, since `2` is the correct index of `b` in the list `['a', 'b']`.

### To Reproduce

.

### OS:

Linux

### DuckDB Version:

1.1.0

### DuckDB Client:

Python

### Hardware:

i5, AMD64

### Full Name:

Soeren Wolfers

### Affiliation:

G-Research

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Not applicable - the reproduction does not require a data set

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by ErikJansenIRefact** [comment]
Seems consistent behaviour?

In the first iteration of list_reduce the values 'a' and 'b' are taken (hence i = 1).
The result is this query: SELECT list_reduce(['a', 'b','c'], (x, y, i) -> x || y || i);

ab1c2.

So in fact "i" is not the index in the list but the iterator count of list reduce.

**Comment by soerenwolfers** [comment]
The docs say "extra parameter that represents the index of the current element", not "extra parameter that represents the iterator count".

`list_reduce` should eventually take an initial state argument. I think it'd be inconsistent with any phrasing of "extra parameter .. index in the list" if without that argument `(a, b, 1)`  is the first argument while with it `(init, a, 1)` is the first argument. 

Then again, documentation can be changed and I'm happy to do that if I get confirmation that the current behavior is preferred.

**Comment by ErikJansenIRefact** [comment]
I agree with the initial state argument and then index is indeed the current element.

**Comment by soerenwolfers** [comment]
Duplicate of https://github.com/duckdb/duckdb/issues/14619
